pybamm/models/submodels/equivalent_circuit_elements/ocv_element.py

```diff
@@ -56,7 +56,7 @@ def set_initial_conditions(self, variables):
 
     def set_events(self, variables):
         soc = variables["SoC"]
-        self.events = [
+        self.events += [
             pybamm.Event("Minimum SoC", soc),
             pybamm.Event("Maximum SoC", 1 - soc),
         ]
```

Comment:
Every time this is called, it would append these events. I think that is a bit counter intuitive to what "set_events()" would typically mean. I would expect a set function to replace what is currently there, not add to it. Can we remove this?

Reply:
I suggested this change as I was afraid that otherwise, if this submodel was processed after some events had already been defined it would erase them. Not sure what's the best way forward.

Reply:
This adds a side effect to what would be expected to be a simple setter. At minimum it should be changed to `append_events()` or `add_events()`. Was there an actual bug that this was fixing or just not sure if it would cause issues?

Reply:
It was not sure, but before only the OCV component defined events, while now there are the OCV and the diffusion components.